### DISCUSSION TITLE
Charlie Marsh - Ruff

### DISCUSSION THREAD

**Discussion by hipstersmoothie** [devtools-fm/devtools.fm#156] [Announcements]
## Background

- https://twitter.com/charliermarsh
- https://github.com/charliermarsh/
- https://www.crmarsh.com
- https://github.com/charliermarsh/ruff

### Background

- [ ] Where did you get your start and how did that lead you to python tooling?

### Ruff

- [ ] What inspired you to build ruff? What did the python linting space look like before?
- [ ] What other tools did you take inspiration from?
- [ ] Where did you start? (from zero or a fork) Did you build your own ast?
- [ ] What are some technical decisions you made to facilitate adoption early on?
- [ ] It seems like ruff blew up overnight an so many huge repos are using it. How did that happen? Has a community formed and started contributing?
- [ ] How did you make it so fast? It can't all just be the rust code. 
- [ ] Are there things that ruff can do that other linters can't? (other than being near instant) the type migrations seems like a huge win
- [ ] What are some challenges that have come up building a tool in one language for a different language?
- [ ] Ruff doesn't support plugins, how do you plan to implement that if at all?

**Comment by charliermarsh** [comment]
These are great!

**Comment by charliermarsh** [comment]
Maybe not totally conventional but in tooltips I wanna talk a bit about some of the other tools I've been learning from, e.g.:

- https://github.com/Boshen/oxc
- https://www.npmjs.com/package/elm-review

**Comment by charliermarsh** [comment]
> It seems like ruff blew up overnight an so many huge repos are using it. How did that happen? Has a community formed and started contributing?

There are a lot of sub-topics we could talk about here, like:

- The initial launch
- Early engagement / adopters
- Ongoing community-building
- Technical / design decisions to facilitate adoption

But not sure how much we want to focus on the tool itself (What can Ruff do? How does it work?) vs. the journey (How did it come to be?). I like talking about both.

**Comment by hipstersmoothie** [comment]
https://marketplace.visualstudio.com/items?itemName=usernamehw.errorlens
https://github.com/ds300/lazyrepo

**Comment by just-be-dev** [comment]
- https://github.com/hmans/miniplex
- https://github.com/cozodb/cozo

**Comment by charliermarsh** [comment]
- https://maturin.rs/
